pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -174,11 +179,18 @@ def on_save_checkpoint(self) -> Dict:
         state: Optional[MergedIteratorState] = getattr(self._data_fetcher.dataloader_iter, state_to_save, None)
         if state:
             state_dict["dataloader_state_dict"] = asdict(state)
+        state_dict["dataloader_state_dict"] = _collect_states_on_rank_zero_over_collection(
+            state_dict["dataloader_state_dict"]
+        )
```

Comment:
nit: access the dictionary just once
```suggestion
        if state:
            state = asdict(state)
        state = _collect_states_on_rank_zero_over_collection(state)
        state_dict["dataloader_state_dict"] = state
```

Reply:
Done !